resources/saas-boost-public-api.yaml

```diff
@@ -125,6 +134,68 @@ Parameters:
     Description: User Service auth token Lambda ARN
     Type: String
 Resources:
+  AuthorizerExecRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub sb-${Environment}-authorizer-role-${AWS::Region}
+      Path: '/'
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: !Sub sb-${Environment}-authorizer-policy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:PutLogEvents
+                Resource: !Sub arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:*:log-stream:*
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                  - logs:DescribeLogStreams
+                Resource:
+                  - !Sub arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:*
+  AuthorizerLogs:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/lambda/sb-${Environment}-authorizer
+      RetentionInDays: 30
+  AuthorizerLambda:
+    Type: AWS::Lambda::Function
+    DependsOn: AuthorizerLogs
+    Properties:
+      FunctionName: !Sub sb-${Environment}-authorizer
+      Role: !GetAtt AuthorizerExecRole.Arn
+      Runtime: java11
+      Timeout: 30
+      MemorySize: 1024
+      Handler: com.amazon.aws.partners.saasfactory.saasboost.ApiGatewayAuthorizer
+      Code:
+        S3Bucket: !Ref SaaSBoostBucket
+        S3Key: !Sub ${LambdaSourceFolder}/Authorizer-lambda.zip
+      Layers:
+        - !Ref SaaSBoostUtilsLayer
+      Environment:
+        Variables:
+          SAAS_BOOST_ENV: !Ref Environment
+          JAVA_TOOL_OPTIONS: '-XX:+TieredCompilation -XX:TieredStopAtLevel=1'
+          IDENTITY_PROVIDER: COGNITO
```

Comment:
Would be handy to extract this to a parameter now to simplify future changes.

Reply:
It will be a parameter in the next PR. Just trying to keep these PRs bite sized.